crates/pipeline-manager/src/cluster_monitor.rs

```diff
@@ -0,0 +1,355 @@
+use crate::config::CommonConfig;
+use crate::db::error::DBError;
+use crate::db::storage::Storage;
+use crate::db::storage_postgres::StoragePostgres;
+use crate::db::types::monitor::{MonitorStatus, NewClusterMonitorEvent};
+use crate::error::source_error;
+use async_trait::async_trait;
+use feldera_observability::ReqwestTracingExt;
+use std::{sync::Arc, time::Duration};
+use tokio::sync::Mutex;
+use tracing::{error, info};
+use uuid::Uuid;
+
+/// Interval at which the monitor occurs to check the current status.
+const MONITOR_INTERVAL: Duration = Duration::from_secs(60);
```

Comment:
I still think this should be more frequent, e.g., 10s. When something goes wrong, we want the status to change in near-real-time. The overhead of polling every 10s should still be negligible.